r/R/dplyr-funcs-datetime.R

```diff
@@ -300,6 +314,34 @@ register_bindings_duration <- function() {
 
     build_expr("cast", x, options = cast_options(to_type = duration(unit = "s")))
   })
+  register_binding("decimal_date", function(date) {
+    y <- build_expr("year", date)
+    start <- call_binding("make_datetime", year = y, tz = "UTC")
+    sofar <- call_binding("difftime", date, start, units = "secs")
+    total <- call_binding(
+      "if_else",
+      build_expr("is_leap_year", date),
+      call_binding("as.integer64", 31622400L),  # number of seconds in a leap year (366 days)
+      call_binding("as.integer64", 31536000L)   # number of seconds in a regular year (365 days)
```

Comment:
Could these be `Expression$scalar(31622400L)`?

Reply:
You are 100% right. The newer implementation is quicker. By how much - it depends, but always quicker. In general it takes half the time.
![image](https://user-images.githubusercontent.com/13176361/160692542-c110b82a-19f2-43e2-938c-5e60e06fc7a2.png)

Reply:
What kind of scale are those numbers on? Could you post or link to the code you used to run that, I would be interested to see how you got there!

Reply:
https://gist.github.com/dragosmg/673a5b23a434adbe0d4e972d7b8a4620

Reply:
```r
summary(test1, relative = TRUE)
#>  A tibble: 2 × 13
#>   expression           min median `itr/sec` mem_alloc `gc/sec` n_itr  n_gc total_time result   memory     time       gc      
#>   <bch:expr>         <dbl>  <dbl>     <dbl>     <dbl>    <dbl> <int> <dbl>   <bch:tm> <list>   <list>     <list>     <list>  
#> 1 new_implementation  1      1         1.40      1        1.03    94     6      1.32s <tibble> <Rprofmem> <bench_tm> <tibble>
#> 2 old_implementation  1.46   1.46      1         1.27     1       92     8      1.81s <tibble> <Rprofmem> <bench_tm> <tibble>

test1
#>  A tibble: 2 × 13
#>   expression              min  median `itr/sec` mem_alloc `gc/sec` n_itr  n_gc total_time result   memory     time       gc      
#>   <bch:expr>         <bch:tm> <bch:t>     <dbl> <bch:byt>    <dbl> <int> <dbl>   <bch:tm> <list>   <list>     <list>     <list>  
#> 1 new_implementation   12.7ms  13.3ms      71.1    76.1KB     4.54    94     6      1.32s <tibble> <Rprofmem> <bench_tm> <tibble>
#> 2 old_implementation   18.5ms  19.4ms      50.8    96.9KB     4.42    92     8      1.81s <tibble> <Rprofmem> <bench_tm> <tibble>

Reply:
I don't have any reason to believe this would be different at larger scales, but for benchmarks like these we probably should use larger sets of data than a handful of rows. It depends on what we're looking at, but 10k-100k rows is more reasonable for something like this